lib/api/completeMultipartUpload.js

```diff
@@ -114,29 +115,56 @@ function completeMultipartUpload(authInfo, request, log, callback) {
 
     async.waterfall([
         function waterfall1(next) {
+            const metadataValParams = {
+                objectKey,
+                authInfo,
+                bucketName,
+                // Required permissions for this action
+                // at the destinationBucket level are same as objectPut
+                requestType: 'objectPut',
+                log,
+            };
+            services.metadataValidateAuthorization(metadataValParams,
```

Comment:
We were checking the auth, but the order was switched around.
(This way, if we encounter error with auth for source dest, we would have the bucket for dest already to gather cors headers from).
I checked with Lauren about this and she seemed to think it would be fine.